docs/4.4/api-reference.md

```diff
@@ -818,14 +820,99 @@ if err := client.DeleteAccessRequest(ctx, accessReqID); err != nil {
+
+```go
+// This will start the an automatic rotation that schedules each phase of the rotation in equal increments.
+gracePeriod := time.Hour * 24
+req := auth.RotateRequest{
+  Mode: services.RotationModeAuto
+  GracePeriod: &gracePeriod // defaults to 48 hours
+  Type: services.UserCA // Leave empty to target UserCA and HostCA
+}
+if err := client.RotateCertAuthority(req); err != nil {
+  return err
+}
+```
+
+In `auto` mode, the rotation is scheduled to run in a series of phases. These phases occur in the order below, unless you set a custom schedule, or run each phase manually.
+
+```go
+// RotationPhaseStandby is the default state for a CA's rotation object - no rotation operations are underway.
+RotationPhaseStandby = "standby"
+// RotationPhaseInit is a phase of the rotation when the new certificate authority is issued, but its not used yet. It is necessary for remote trusted clusters to fetch the new certificate authority, otherwise the new clients will reject it.
+RotationPhaseInit = "init"
+// RotationPhaseUpdateClients is a phase of the rotation when client credentials will have to be updated and reloaded, but servers will use and respond with old credentials because clients (grace period).
+RotationPhaseUpdateClients = "update_clients"
+// RotationPhaseUpdateServers is a phase of the rotation when servers will have to reload and should start serving TLS and SSH certificates signed by new CA (retrieved in previous phase).
+RotationPhaseUpdateServers = "update_servers"
+// RotationPhaseRollback means that the rotation is rolling back to the old certificate authority, may occur in the case of a rotation failure.
+RotationPhaseRollback = "rollback"
+// The rotation will end in Standby after the rotation completes
+```
+
+You can also rotate certificates with a custom schedule. The default schedule gives the `Init`, `UpdateClients`, and `UpdateServers` phases each 1/3 of the total grace period to complete. You may want to extend the length of one phase in the schedule without having a ridiculously long grace period, since long grace periods present a possible vulnerability
+
+```go
+// This will automate your CA rotation with the custom schedule. Use with caution, each phase should have times extra time to ensure they complete in less than optimal situations.
+clock := clockwork.NewRealClock()
+if err := client.RotateCertAuthority(auth.RotateRequest{
+  Mode: services.RotationModeAuto,
+  Schedule: &services.RotationSchedule{
+    // 1 hour for Init
+    UpdateClients: clock.Now().UTC().Add(time.Hour).UTC(),
+    // 4 hours for UpdateClients
+    UpdateServers: clock.Now().UTC().Add(time.Hour * 5).UTC(),
+    // 2 hours for UpdateServers
+    Standby: clock.Now().UTC().Add(time.Hour * 7).UTC(),
+  },
+}); err := client.RotateCertAuthority(req); err != nil {
+  return err
+}
+```
```

Comment:
Setting a custom schedule seems like it has weird behavior, but I haven't tested it extensively. I can remove this section if this is not a desired use case.